compiler/testSuite/semtype/recursive-t.bal

```diff
@@ -0,0 +1,14 @@
+type R1 record {| R2 f; |};
+
+type R2 record {| R1 f; |};
+
+// @type R1 <> L
+type L string[];
```

Comment:
Do we currently fail this? They are different basic types...

Reply:
Currently `// @type R1 < L` evaluates to `true`.  `R1^~L` becomes `R1` and when we check `isEmpty(R1)` that becomes `true` due to `if res == () { return true}` condition as in R1 we get the same bdd again when checking the emptiness.

Reply:
So R1 and R2 are treated as equivalent to `never`?

Reply:
At the moment that's the behaviour.

Reply:
OK, so that's the real issue then.

Reply:
Can you update issue accordingly?

Reply:
Simple test case is then:

```
type L [L];
type N never;
// Should we have N == L
// @type N < L
```